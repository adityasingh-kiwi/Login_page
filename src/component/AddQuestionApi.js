
import React, { Component } from "react";
import axios from "axios";
const httpOptions = {
    headers:{
        'Content-Type': 'application/json',
        Authorization: 'my-token',
    }
};




export default class RequestService extends Component{


    constructor(props) {
        super(props);
       
       this.baseApiUrl = 'https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/loginpage.json';
 }

     static getQuestionOne() {
          try {
            
            return axios.get("https://question-store-fccce-default-rtdb.firebaseio.com/question.json");
         
     } catch (error) {
         
     }
     
         
    }

    static getAnswerOne() {
        try {
          
          return axios.get("https://score-view-fd276-default-rtdb.firebaseio.com/score.json");
       
   } catch (error) {
       
   }
   
       
  }
    1   
    static getWithParams(endPoint, params) {
        return axios.get(`${this.baseApiUrl}${endPoint}`, { params });
    }
    static  post(endPoint, data, formData = false) {
        if (formData) {
            httpOptions.headers["Content-Type"] = 'application/x-www-form-urlencoded';
        }
        return axios.post(`${this.baseApiUrl}${endPoint}`, data,httpOptions);
    }

    static putOne(endPoint, data) {
    

        try {
            if(typeof data == "object"){
                httpOptions.headers["Content-Type"] = 'application/x-www-form-urlencoded';
                 axios.put("/https://question-store-fccce-default-rtdb.firebaseio.com/question.json/question.json/${endPoint}.json", data, httpOptions)
                 .then((response)=>{
                     
                    
                       return response;


                 }).catch((err)=>{
                    console.log("err---------------------->>>",err)

                 })

                 
            }
        } catch (error) {
            console.log("error---------------------->>>",error)
            
        }
       
       
    }

   
   

}
