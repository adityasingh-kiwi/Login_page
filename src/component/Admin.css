.button-add-question {
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
  }

  .button-view-score {
    background-color: #008CBA;
    border: none;
    color: white;
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 4px 2px;
    cursor: pointer;
  }
  .background-admin{
      position: relative;
      /* top: 250px;
      left:40%; */
      padding: 200px 451px;
      
  }
  .background-image{
    background-image: url("/home/kiwitech/Desktop/Project/login_page/src/image/Survey.jpg");
    height: 611px;
    background-size: cover;
  }

   .addquestion-question{
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }
  .addquestion-choices{
    width: 100%;
    padding: 15px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }
  .addquestion-answer{
    width: 100%;
    padding: 15px 20px;
    margin: 8px 0;
    box-sizing: border-box;
  }