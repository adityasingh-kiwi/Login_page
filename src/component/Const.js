

export const genericFalse= false;
export const genericTrue= true;
export const genericAdminEmailid="admin@gmail.com";
export const genericValueZero = 0;
export const genericValueOne=1;