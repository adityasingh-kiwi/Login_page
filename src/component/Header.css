.header {
    overflow: hidden;
    background-color:#363945;
      padding: 20px 10px;
      
  }

  .Background_image{
    background-image: url("/home/kiwitech/Desktop/Project/login_page/src/image/Survey.jpg");
    background-size: cover;
    overflow: hidden;
  }
  

  
  .header a {
    float: left;
    color: black;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px;
  }
  
  .header a.logo {
    font-size: 25px;
    color: #fff;
    font-weight: bold;
  }
  
  .header a:hover {
    background-color: #ddd;
    color: black;
  }
  

  .header button.active {
    background-color: #2b3e50;
    color:rgb(0, 26, 255);
    border-radius: 18px;
    height: 70px;
    width: 200px;
  }
  .header button.logout {
    background-color: #003366;
    color: red;
    border-radius: 8px;
    height: 50px;
    width: 120px;
  }

  

  .header-right {
    float: right;

  }


  
  

 .profileImage{
   width: 50px !important;
   height: 50px !important;
 }

 .text_colour{
  color:white;
  cursor: pointer;

 }
 




              