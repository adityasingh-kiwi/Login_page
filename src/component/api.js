
import React, { Component } from "react";
import axios from "axios";
const httpOptions = {
    headers:{
        'Content-Type': 'application/json',
        Authorization: 'my-token',
    }
};


export default class RequestService extends Component{


    constructor(props) {
        super(props);
       
       this.baseApiUrl = 'https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/loginpage.json';
 }

     static getOne() {
          try {
            
            return axios.get("https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/loginpage.json");
         
     } catch (error) {
         
     }
         
    }
    static getOnlyOne(id) {
        
      return axios.get(`https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/${id}.json`);
  }
    static getWithParams(endPoint, params) {
        return axios.get(`${this.baseApiUrl}${endPoint}`, { params });
    }
    static  post(endPoint, data, formData = false) {
        if (formData) {
            httpOptions.headers["Content-Type"] = 'application/x-www-form-urlencoded';
        }
        return axios.post(`${this.baseApiUrl}${endPoint}`, data,httpOptions);
    }

    static putOne(endPoint, data) {
    

        try {
            if(typeof data == "object"){
                httpOptions.headers["Content-Type"] = 'application/x-www-form-urlencoded';
                 axios.put("https://login-page-9c380-default-rtdb.asia-southeast1.firebasedatabase.app/loginpage.json/${endPoint}.json", data, httpOptions)
                 .then((response)=>{
                     
                    
                       return response;


                 }).catch((err)=>{
                    console.log("err---------------------->>>",err)

                 })

                 
            }
        } catch (error) {
            console.log("error---------------------->>>",error)
            
        }
       
       
    }

    static delete(endPoint) {
        return axios.delete(`${this.baseApiUrl}${endPoint}`);
    }

   

}
