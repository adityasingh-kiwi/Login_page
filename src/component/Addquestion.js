import React from 'react'
import './Admin.css';
import { useState } from 'react/cjs/react.development';
import { Modal } from 'antd';
import { v4 as uuidv4 } from 'uuid';


export const Addquestion = ( { isopen ,isOpenClose}) => {
    const [isModalVisible, setIsModalVisible] = useState(false);
    const [dataQuestion,setDataQuestion]=useState({
        question:"",
        choices:[{
            id:uuidv4(),
            value:""
        }],
        answers:[{
            id:uuidv4(),
            value:""
        }]
    })




    const handleQuestionAnswer = (e) => {
        let modifiedSetData = {...dataQuestion}

        modifiedSetData.choices.forEach(element => {
            if(element.id == e.target.id){
                element.value = e.target.value
            }
            
        });

        setDataQuestion(modifiedSetData)

     }

const handleReset =()=>{
    setDataQuestion({
        question:"",
        choices:[{
            id:uuidv4(),
            value:""
        }],
        answers:[{
            id:uuidv4(),
            value:""
        }]
    })
}

    const handleCancel = () => {
        handleReset();
        isOpenClose(false);
    };


    const handleOk = async () => {
        isOpenClose(false);

       try {
        
        const rest = await fetch("https://question-store-fccce-default-rtdb.firebaseio.com/question.json", {
          method: "POST",
          headers: {
            "Content-type": "application/json",

          },
          body: JSON.stringify({
            dataQuestion

          })
        })

        if(rest.status == 200){
            handleReset();

        }
           
       } catch (error) {
           
       }
        
    };

    const addChoices =()=>{
        
        let modifiedSetData = {...dataQuestion}
        console.log("modifiedSetData------------>>",modifiedSetData);
        modifiedSetData && modifiedSetData.choices.push({
            id:uuidv4(),
            value:""
        })
        
        setDataQuestion(modifiedSetData)

    }

    const handleaddAnswers =()=>{
        let modifiedSetData = {...dataQuestion}
        console.log("modifiedSetData------------>>",modifiedSetData);
        modifiedSetData && modifiedSetData.answers.push({
            id:uuidv4(),
            value:""
        })

        setDataQuestion(modifiedSetData)
    }

    const handleAnswer =(e)=>{
        let modifiedSetData = {...dataQuestion}

        modifiedSetData.answers.forEach(element => {
            if(element.id == e.target.id){
                element.value = e.target.value
            }
            
        });

        setDataQuestion(modifiedSetData)



    }

    const handleQuestion =(e)=>{
        let modifiedSetData = {...dataQuestion}
           modifiedSetData.question=e.target.value

        setDataQuestion(modifiedSetData)

    }
    console.log("dataQuestion-------------->>",dataQuestion)
    return (

        <div>
             <Modal title="ADD THE QUESTION,Choices AND ANSWERS" visible={isopen} onOk={handleOk} onCancel={handleCancel}>
                <div className="form">
                    <form >
                        <div>
                            Question:<input type="text" name="question" className="addquestion-question" type="addquestion-question" value={dataQuestion.question} onChange={handleQuestion} placeholder="Write Question" required />
                            </div>
                            <div>
                                
                                <button onClick={addChoices}>Add Choices</button>
                            </div>
                            <div>
                                {dataQuestion.choices && dataQuestion.choices.map((element,index)=>{
                                    return(
                                        <input type="text"  id ={element.id} name="choices" className="addquestion-choices" type="addquestion-choices" value={dataQuestion.choices[index].value} onChange={handleQuestionAnswer} placeholder="Write the Choices" required />
                                    )
                                })}
                            
                            </div>
                        
                        




                    </form>
                </div>
            </Modal>
        </div>
    )
}
