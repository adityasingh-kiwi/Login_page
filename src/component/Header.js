import React, { useState, useEffect} from 'react'
import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import './Header.css';

import { useNavigate } from "react-router-dom";
import ModalOne from "./Modal"
import RequestService from "./api";


const Header = () => {

    const [getdata, setGetdata] = useState({email:" ",id:" "})


    const [modalState, handleModalState] = useState(false);
    const handleUserDetails = () => {

        handleModalState(true);

    }
    
  let Naviagte = useNavigate();
  

  const handleAdmin=()=>{
    Naviagte("/admin");
  }


    const fetchAllRequest = async ()=>{
        
        let res = await RequestService.getOne()
      
         let userEmail = localStorage.getItem("data")
         if(res.status == 200){
            console.log("userEmail------------->>>",userEmail);
          
             for(let property in res["data"]){
    

                if(res["data"][property]["userEmail"] == userEmail ){
                    
                   let update =  await RequestService.getOnlyOne(property);
                   
                    setGetdata(
                        {
                            ...getdata,
                            email:res["data"][property]["userEmail"],
                            id:property
                    } 
                    );
                }
              
             }
         }

        
    }
     useEffect(() => {
        fetchAllRequest();
    }, [])

     

    const handleOpenClose =(data)=>{
    

        if(!data){
            handleModalState(false);
        }
        
    }
    
    const handleLogout=()=>{
        Naviagte("/");

    }





    return (
        

        <div className="header">
            <a href="#default" className="logo">Naviagtion</a>

            <div className="header-right" style={{display:"flex"}}>
                
                <div className="col-md-4">

                    <img className="rounded-circle z-depth-2 profileImage" alt="20x20" src="https://mdbootstrap.com/img/Photos/Avatars/img%20(31).jpg"
                        data-holder-rendered="true"/>

                </div>
               
                <button className="active" onClick={handleUserDetails}>{getdata.email}</button>
                &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
                <button className="logout" onClick={handleLogout}>Logout</button>
                &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;

                {getdata.email==="admin@gmail.com" ?   <button className="logout"onClick={handleAdmin}>Admin</button> :null}
               

                <div className="background">
             



            </div>

            
            {
                <ModalOne  getdata={getdata}isopen={modalState} isOpenClose ={(data)=>handleOpenClose(data)}/>
            }
            <img className="mask-custom"></img>

        </div>
        </div>
    )
}



export default Header
