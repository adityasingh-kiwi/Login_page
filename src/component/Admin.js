import React ,{useState} from 'react';
import './Admin.css';
import { Addquestion } from "./Addquestion";
import { useNavigate } from "react-router-dom";
import { genericFalse,genericTrue } from './Const';



 const Admin = () => {

    const [isopen ,setisopen] = useState(genericFalse);


    const handleOpenModal =()=>{
        setisopen(genericTrue) 
    }

    const isOpenClose =(data) =>{
        setisopen(data)
    }
    let Naviagte = useNavigate();
    const handleView=()=>{
        Naviagte("/viewscore");


    }


    return (
        <div className="background-image"> 
        <div className="background-admin">
            <button className="button-add-question" onClick={handleOpenModal}> Add Question</button>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <button className="button-view-score" onClick={handleView}>View Score</button>
        </div>

        <Addquestion   isopen ={isopen} isOpenClose ={(data)=>isOpenClose(data)}/>
        </div>
    )
}
export default Admin
