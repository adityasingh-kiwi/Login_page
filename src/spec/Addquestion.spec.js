import { Addquestion } from "../component/Addquestion";


describe('addQuestion_page',()=>{
    it("Check_handle_change_Addquestion", () => {
        const wrapper = shallow(<Addquestion />);
        const input = wrapper.find(".addQuestion");
        input.simulate("change", { target: { value: "who is the prime minister of india" } });  // 'value' instead of 'num'
       
      });
      it("Check_handle_change_userEmail", () => {
        const wrapper = shallow(<Addquestion />);
        const input = wrapper.find(".addChoices");
        input.simulate("change", { target: { value: "Manmohan singh" } });  
       
      });
      it("Check_handle_change_userEmail", () => {
        const wrapper = shallow(<Addquestion />);
        const input = wrapper.find(".addAnswer");
        input.simulate("change", { target: { value: "Modi" } }); 
       
      });

    it('addQuestion_button check',()=>{
         let wrapper= shallow(<Addquestion/>);
         expect(wrapper.exists('.addChoices_button')).toEqual(true);
         expect(wrapper.length).toEqual(1)
         wrapper.find('.addChoices_button').simulate('click');
    
    })
    it('viewScore_button check',()=>{
     let wrapper= shallow(<Addquestion/>)
     expect(wrapper.exists('.addAnswer_button')).toEqual(true);
     expect(wrapper.length).toEqual(1)
     wrapper.find('.addAnswer_button').simulate('click');
 
 })
 it('Snapshot_AddQuestion',()=>{
  let wrapper=shallow(<Addquestion/>)
  expect(wrapper).toMatchSnapshot();


})
it('addd_button check',()=>{
  let wrapper= shallow(<Addquestion/>)
  expect(wrapper.exists('#addAnswer')).toEqual(true);
  expect(wrapper.length).toEqual(1)
  wrapper.find('#addAnswer').simulate('click');
  expect(
    wrapper.find('#addAnswer').prop('defaultValue'),
    ).toBeTruthy();

})
it('addchoices_button check',()=>{
  let wrapper= shallow(<Addquestion/>);
  expect(wrapper.exists('#addChoices')).toEqual(true);
  expect(wrapper.length).toEqual(1)
  wrapper.find('#addChoices').simulate('click');
  expect(
    wrapper.find('#addChoices').prop('defaultValue'),
    ).toBeTruthy();

})
it("Check_handle_Question_change_Function", () => {
  const wrapper = shallow(<Addquestion />);
  const input = wrapper.find("#add_question");
  input.simulate("change", { target: 
    { name:"question",
      value: "who is the prime minister of india"
     }
    }) 

    expect(
      wrapper.find('#add_question').prop('defaultValue'),
      ).toBeTruthy();
      
    
    
    
    

   
});
it("Check_handle_Question_change_Function", () => {
  const wrapper = shallow(<Addquestion />);
  const input = wrapper.find("#add_question");
  input.simulate("change", { target: 
    { name:"question",
      value: "who is the prime minister of india"
     }
    }) 

    expect(
      wrapper.find('#add_question').prop('defaultValue'),
      ).toBeTruthy();
      
    
    
    
    

   
});
it("Check_handle_Question_change_Function", () => {
  const wrapper = shallow(<Addquestion />);
  const input = wrapper.find("#choices");
  input.simulate("change", { target: 
    { name:"choices",
      value: "india"
     }
    }) 

    expect(
      wrapper.find('#choices').prop('defaultValue'),
      ).toBeTruthy();
      
    
    
    
    

   
});
it("Check_handle_Question_change_Function", () => {
  const wrapper = shallow(<Addquestion />);
  const input = wrapper.find("#answers");
  input.simulate("change", { target: 
    { name:"answer",
      value: "india"
     }
    }) 

    expect(
      wrapper.find('#answers').prop('defaultValue'),
      ).toBeTruthy();
      
    
    
    
    

   
});



 
 })