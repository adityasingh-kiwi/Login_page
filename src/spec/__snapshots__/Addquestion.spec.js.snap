// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addQuestion_page Snapshot_AddQuestion 1`] = `
<div>
  <Modal
    confirmLoading={false}
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="ADD THE QUESTION,Choices AND ANSWERS"
    visible={false}
    width={520}
  >
    <div
      className="form"
    >
      <form>
        <div>
          <div
            className="addQuestion"
          >
            Question:
            <input
              className="addquestion-question"
              defaultValue={true}
              id="add_question"
              name="question"
              onChange={[Function]}
              placeholder="Write Question"
              required={true}
              type="addquestion-question"
              value=""
            />
          </div>
          <div
            className="addChoices_button"
          >
            <button
              defaultValue={true}
              id="addChoices"
              onClick={[Function]}
            >
              Add Choices
            </button>
          </div>
          <div
            className="addChoices"
          >
            <input
              className="addquestion-choices"
              defaultValue={true}
              id="choices"
              name="choices"
              onChange={[Function]}
              placeholder="Write the Choices"
              required={true}
              type="addquestion-choices"
              value=""
            />
            <div
              className="addAnswer"
            >
              <input
                className="addquestion-answer"
                defaultValue={true}
                id="answers"
                name="choices"
                onChange={[Function]}
                placeholder="Write the Answers"
                required={true}
                type="addquestion-answer"
                value=""
              />
            </div>
            <div
              className="addAnswer_button"
            >
              <button
                defaultValue={true}
                id="addAnswer"
                onClick={[Function]}
              >
                Add Answers
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </Modal>
</div>
`;
